internal/services/loganalytics/client/client.go

```diff
@@ -27,6 +28,7 @@ type Client struct {
 	StorageInsightsClient      *storageinsights.StorageInsightsClient
 	QueryPackQueriesClient     *querypackqueries.QueryPackQueriesClient
 	WorkspacesClient           *workspaces.WorkspacesClient
+	FeatureWorkspaceClient     *featureWorkspaces.WorkspacesClient // 2022-10-01 API version does not contain sharedkeys related API, so we keep two versions SDK of this API
```

Comment:
Can't we upgrade the entire SDK instead?

Reply:
I may did not get your point, you mean upgrade "querypackquries", "storageinsights" and etc... in "loganalytics" or remove the old workspaces SDK?
for the second one, we cant do it for now because the newer SDK does not contain sharedkeys functions used here https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/loganalytics/log_analytics_workspace_resource.go#L358